internal/resource/resource_plugin.go

```diff
@@ -70,15 +85,54 @@ func (r *Resource) Process(ctx context.Context, msg *bus.Message) {
 		return
 
 	case bus.DeletedInstancesTopic:
-		instanceList, ok := msg.Data.([]*v1.Instance)
+		instanceList, ok := msg.Data.([]*mpi.Instance)
 		if !ok {
-			slog.ErrorContext(ctx, "Unable to cast message payload to []*v1.Instance", "payload", msg.Data)
+			slog.ErrorContext(ctx, "Unable to cast message payload to []*mpi.Instance", "payload", msg.Data)
 		}
 		resource := r.resourceService.DeleteInstances(instanceList)
 
 		r.messagePipe.Process(ctx, &bus.Message{Topic: bus.ResourceUpdateTopic, Data: resource})
 
 		return
+	case bus.WriteConfigSuccessfulTopic:
+		data, ok := msg.Data.(model.ConfigApply)
+		if !ok {
+			slog.ErrorContext(ctx, "Unable to cast message payload to instanceID string", "payload", msg.Data)
+		}
+		err := r.resourceService.Apply(ctx, data.InstanceID)
+		if err != nil {
+			slog.Error("errors found during config apply, sending failure status", "err", err)
+			response := &mpi.DataPlaneResponse{
+				MessageMeta: &mpi.MessageMeta{
```

Comment:
MessageMeta could be done in one place and reused outside of the err condition. Probably simpler to keep the err check to manage conditions specific to the error and then have code outside of that like 
```			
r.messagePipe.Process(ctx, &bus.Message{Topic: bus.DataPlaneResponseTopic, Data: response})
r.messagePipe.Process(ctx, &bus.Message{Topic: bus.ConfigApplyFailedRequestTopic, Data: data.InstanceID})
```